internal/client/cluster/dataprotection/dataprotection_resource.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package dataprotectionclient
+
+import (
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/transport"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/cluster/dataprotection"
+	"net/url"
+	"strconv"
+)
+
+const (
+	apiVersionAndGroup = "v1alpha1/clusters"
+	dataProtectionPath = "dataprotection"
+)
+
+// New creates a new cluster resource service API client.
+func New(transport *transport.Client) ClientService {
+	return &Client{Client: transport}
+}
+
+/*
+Client for credentials resource service API.
+*/
+type Client struct {
+	*transport.Client
+}
+
+// ClientService is the interface for Client methods.
+type ClientService interface {
+	DataProtectionResourceServiceCreate(request *dataprotectionmodels.VmwareTanzuManageV1alpha1ClusterDataprotectionCreateDataProtectionRequest) (*dataprotectionmodels.VmwareTanzuManageV1alpha1ClusterDataprotectionCreateDataProtectionResponse, error)
+
+	DataProtectionResourceServiceDelete(fn *dataprotectionmodels.VmwareTanzuManageV1alpha1ClusterDataprotectionFullName, destroyBackups bool) error
+
+	DataProtectionResourceServiceList(clusterName string) (*dataprotectionmodels.VmwareTanzuManageV1alpha1ClusterDataprotectionListDataProtectionsResponse, error)
```

Comment:
Data Protection also support listing all clusters with data protection enabled via wildcard.
Are we not supporting similar in terraform?

Reply:
Currently the data protection is based on an individual cluster an this is how I understood the scope of the task and therefore it is implemented this way.
Who can confirm the scope of the task?

Reply:
@ramya-bangera @shobha2626 Are we supporting listing anywhere else in terraform?

Reply:
As mentioned by @GilTeraSky, Data protection resource is based on individual cluster. That case, we don't need list implementation.

Reply:
Actually I removed the datasource for enable data protection because initially we thought users will be using this data source to get the backup location which mentioned to be deperecated and requested to be removed from the resource.